daemon/src/oracle.rs

```diff
@@ -252,14 +259,37 @@ impl Actor {
         self.update_pending_attestations(ctx);
     }
 
-    async fn handle_new_attestation_fetched(&mut self, msg: NewAttestationFetched) {
+    async fn handle_new_attestation_fetched(&mut self, msg: NewAttestationFetched) -> Result<()> {
         let NewAttestationFetched { id, attestation } = msg;
 
         tracing::info!("Fetched new attestation for {id}");
 
+        let mut conn = self.db.acquire().await?;
+
+        for id in db::load_all_cfd_ids(&mut conn).await? {
+            let cfd = try_continue!(load_cfd(id, &mut conn).await);
+            let event = try_continue!(cfd
+                .decrypt_cet(&attestation.clone())
+                .context("Failed to decrypt CET using attestation"));
+
+            if let Some(event) = event {
+                // Note: ? OK, because if the actor is disconnected we can fail the loop
+                if let Err(e) = self
+                    .process_manager
+                    .send(process_manager::Event::new(event.clone()))
+                    .await?
+                {
+                    tracing::error!("Sending event to process manager failed: {:#}", e);
+                }
+            }
+        }
+
         let _: Result<(), xtra::Disconnected> =
             self.attestation_channel.send_async_safe(attestation).await;
```

Comment:
If you are already calling `decrypt_cet` above, sending the attestation through this channel _as well_ ends up calling `decrypt_cet` a second time. Remember that this channel is really just the `{maker,taker}_cfd::Actor` and all they are doing with the attestation is calling the `cfd_actors::handle_oracle_attestation` which does exactly the loop above.

However, simply removing this channel is what causes tests to break (because they rely on being able to send an attestation to the `{maker,taker}_cfd::Actor`. (That is what https://github.com/itchysats/itchysats/issues/1148 is about.)

From the description, I take it that your intent was to do https://github.com/itchysats/itchysats/issues/1147 first. For that, this code doesn't need to change and this change should go into your other PR!